### ISSUE TITLE
Vim9: E1004 when using comparison operator in lambda passed to sort()

### ISSUE THREAD

**Issue by lacygoill** [vim/vim#6917]
**Describe the bug**

In Vim9 script, `E1004` is wrongly raised when using a comparison operator in a lambda passed as the second optional argument to `sort()`.

**To Reproduce**

Run this shell command:

    vim -Nu NONE -S <(cat <<'EOF'
        vim9script
        sort([#{lnum: 2}, #{lnum: 1}], {i, j -> i.lnum > j.lnum ? 1 : -1})
            ->assert_equal([#{lnum: 1}, #{lnum: 2}])
    EOF
    )

`E1004` is raised twice:

    E1004: white space required before and after '>'
    E1004: white space required before and after '>'

**Expected behavior**

No error is raised, because – inside the lambda passed to `sort()` – the comparison operator `>` is properly surrounded by whitespace (before and after).

**Environment**

 - Vim version: 8.2 Included patches: 1-1643
 - OS: Ubuntu 16.04.7 LTS
 - Terminal: xterm(359)

**Additional context**

Regression introduced in [8.2.1368](https://github.com/vim/vim/releases/tag/v8.2.1368).

---
Patch adding a test:
```diff
diff --git a/src/testdir/test_vim9_expr.vim b/src/testdir/test_vim9_expr.vim
index 3c16fa48e..12accbe87 100644
--- a/src/testdir/test_vim9_expr.vim
+++ b/src/testdir/test_vim9_expr.vim
@@ -857,6 +857,13 @@ def Test_expr4_vim9script()
     echo line[1] =~ '\w'
   END
   CheckScriptSuccess(lines)
+
+  lines =<< trim END
+    vim9script
+    sort([#{lnum: 2}, #{lnum: 1}], {i, j -> i.lnum > j.lnum ? 1 : -1})
+        ->assert_equal([#{lnum: 1}, #{lnum: 2}])
+  END
+  CheckScriptSuccess(lines)
 enddef
 
 func Test_expr4_fails()
```

**Comment by lacygoill** [comment]
The issue only occurs at the script level.  In a `:def` function, the code works fine:
```vim
vim9script
def Func()
    sort([#{lnum: 2}, #{lnum: 1}], {i, j -> i.lnum > j.lnum ? 1 : -1})
        ->assert_equal([#{lnum: 1}, #{lnum: 2}])
enddef
Func()
```

**Comment by lacygoill** [comment]
Fixed by [8.2.1671](https://github.com/vim/vim/releases/tag/v8.2.1671).
